eliasgierlinger/unguard
# 🧽[like-service, frontend] Refactor code, make PHP SQL Injection exploitable through frontend

* Remove or comment out unnecessary Laravel code (see https://www.reddit.com/r/laravel/comments/du7gdl/is_there_a_basic_minimal_hello_world_laravel/f73b0r7/?utm_source=share&utm_medium=web3x&utm_name=web3xcss&utm_term=1&utm_content=share_button)
* Clean up Laravel code 
    * Use static function, remove double-assignment of variable
* Change the way that liking and unliking is handled
    * Change /post/postid parameter from being path parameter to being query/GET parameter and add optional like_delete and like_post query parameters
    * Remove separate redirecting /like route; use /post with new parameters instead
* Add `zip` to Dockerfile, as it would not build otherwise / would take very long

To test the SQL Injection exploit:
* Open unguard in browser twice, logging in with two separate accounts (admanager and a custom account) - admanager has a known userid of 1
* Like the same post with both accounts
* On the custom account, open http://unguard.kube/ui/post?postId=[POST_ID]&postId=1&like_delete=
* Will return an error, since a separate call failed. Call with SQL injection should have worked nonetheless
* On the admanager account, open http://unguard.kube/ui/post?postId=[POST_ID]
* Admanager's like was removed by custom account
